### PR TITLE
Add run statistics endpoint for products

### PR DESCRIPTION
**Pull request by lamppu**
Please see the individual commits for details.

### CODE REVIEW

File: api/v1/model/src/commonMain/kotlin/OrtRunSummary.kt [hunk 1]
```diff
@@ -31,7 +31,7 @@ data class OrtRunSummary(
      * The unique identifier.
      */
     val id: Long,
-
+
```
Comment (sschuberth) [inline api/v1/model/src/commonMain/kotlin/OrtRunSummary.kt:34]: Nit: This should be undone.

File: core/src/test/kotlin/api/RunsRouteIntegrationTest.kt [hunk 2]
```diff
@@ -1816,11 +1727,16 @@ class RunsRouteIntegrationTest : AbstractIntegrationTest({
 
                 val statistics = response.body<OrtRunStatistics>()
 
-                statistics.issuesCount shouldBe 1
-                statistics.packagesCount shouldBe 2
-                statistics.ecosystems?.size shouldBe 2
-                statistics.vulnerabilitiesCount shouldBe null
-                statistics.ruleViolationsCount shouldBe null
+                with(statistics) {
+                    issuesCount.shouldBeNull()
```
Comment (nnobelis) [inline core/src/test/kotlin/api/RunsRouteIntegrationTest.kt:1731]: _nit_ isn't it `shouldBe null` ?
@sschuberth  right ?

+ other places

Comment (sschuberth) [reply core/src/test/kotlin/api/RunsRouteIntegrationTest.kt:1731]: Actually, it's `should beNull()` 😸

Comment (lamppu) [reply core/src/test/kotlin/api/RunsRouteIntegrationTest.kt:1731]: Just out of curiosity is that a syntax choice, or is there some difference?

Comment (sschuberth) [reply core/src/test/kotlin/api/RunsRouteIntegrationTest.kt:1731]: Just syntax. In general, we prefer infix functions, i.e. those without the need for a `.` before the function call, to make the code read more fluent.

**Pull request merged**